In [1]:
#Import dependencies

import pandas as pd
import numpy as np

In [85]:
# Make a reference to the books.csv file path
filepath = "Resources/purchase_data.csv"

# Import the books.csv file as a DataFrame
df = pd.read_csv(filepath)
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price  
0  Extraction, Quickblade Of Trembling Hands   3.53  
1                          Frenzied Scimitar   1.56  
2                               Final Critic   4.88  
3                                Blindscythe   3.27  
4                                       Fury   1.44

In [86]:
purchase_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 780 entries, 0 to 779
Data columns (total 7 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   Purchase ID  780 non-null    int64  
 1   SN           780 non-null    object 
 2   Age          780 non-null    int64  
 3   Gender       780 non-null    object 
 4   Item ID      780 non-null    int64  
 5   Item Name    780 non-null    object 
 6   Price        780 non-null    float64
dtypes: float64(1), int64(3), object(3)
memory usage: 42.8+ KB


In [87]:
len(df.SN)

780

In [88]:
df.shape

(780, 7)

In [89]:
df.columns

Index(['Purchase ID', 'SN', 'Age', 'Gender', 'Item ID', 'Item Name', 'Price'], dtype='object')

In [90]:
df.count()

Purchase ID    780
SN             780
Age            780
Gender         780
Item ID        780
Item Name      780
Price          780
dtype: int64

In [91]:
df.value_counts()

Purchase ID  SN            Age  Gender  Item ID  Item Name                                  Price
0            Lisim78       20   Male    108      Extraction, Quickblade Of Trembling Hands  3.53     1
536          Siallylis44   20   Male    92       Final Critic                               4.19     1
514          Yasur85       21   Male    21       Souleater                                  1.10     1
515          Haillyrgue51  7    Male    40       Second Chance                              2.52     1
516          Aidain51      39   Female  35       Heartless Bone Dualblade                   3.45     1
                                                                                                    ..
262          Chamadar79    15   Male    172      Blade of the Grave                         3.14     1
263          Aeral43       24   Male    78       Glimmer, Ender of the Moon                 4.40     1
264          Saesrideu94   35   Male    119      Stormbringer, Dark Blade of E

# Player Count

In [92]:
df.SN.nunique()

576

In [93]:
tot_players = df.SN.nunique()

summ1 = pd.DataFrame()
summ1["Total Players"] = [tot_players]

summ1

Total Players
0            576

# Purchasing Analysis (Total)

In [94]:
num_unique_items = df["Item ID"].nunique()
avg_pur_price = df.Price.mean()
tot_pur_price = len(df["Item ID"])
tot_rev = df.Price.sum()

summ2 = pd.DataFrame()
summ2["Number of Unique Items"] = [num_unique_items]
summ2["Average Price"] = [avg_pur_price]
summ2["Number of Purchases"] = [tot_pur_price ]
summ2["Total Revenue"] = [tot_rev]

summ2

Number of Unique Items  Average Price  Number of Purchases  Total Revenue
0                     179       3.050987                  780        2379.77

# Gender Demographics

In [95]:
cols = ["SN", "Age", "Gender"]

players = df.loc[:, cols]
players.head()

SN  Age Gender
0        Lisim78   20   Male
1    Lisovynya38   40   Male
2     Ithergue48   24   Male
3  Chamassasya86   24   Male
4      Iskosia90   23   Male

In [96]:
players = players.drop_duplicates().reset_index(drop=True)
players.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 576 entries, 0 to 575
Data columns (total 3 columns):
 #   Column  Non-Null Count  Dtype 
---  ------  --------------  ----- 
 0   SN      576 non-null    object
 1   Age     576 non-null    int64 
 2   Gender  576 non-null    object
dtypes: int64(1), object(2)
memory usage: 13.6+ KB


In [97]:
players.groupby("Gender").size()

Gender
Female                    81
Male                     484
Other / Non-Disclosed     11
dtype: int64

In [98]:
players.groupby("Gender").size()/len(players)

Gender
Female                   0.140625
Male                     0.840278
Other / Non-Disclosed    0.019097
dtype: float64

In [99]:
player_cnt = players.Gender.value_counts() 
player_perc = (players.Gender.value_counts() / len(players)) * 100

# https://stackoverflow.com/a/18062521
summ3 = pd.concat([player_cnt, player_perc], axis=1)
summ3.columns = ["Total Count", 'Percentage of Players']

summ3

Total Count  Percentage of Players
Male                           484              84.027778
Female                          81              14.062500
Other / Non-Disclosed           11               1.909722

# Purchasing Analysis (Gender)

In [100]:
df.columns

Index(['Purchase ID', 'SN', 'Age', 'Gender', 'Item ID', 'Item Name', 'Price'], dtype='object')

In [101]:
summ4 = df.groupby("Gender").agg({"Purchase ID": "count", "Price":["mean", "sum"]})
summ4.columns = ["Purchase Count", "Average Purchase Price", "Total Purchase Value"]
summ4["Avg Total Purchase per Person"] = summ4["Total Purchase Value"] / summ3["Total Count"]
summ4

Purchase Count  Average Purchase Price  \
Gender                                                          
Female                            113                3.203009   
Male                              652                3.017853   
Other / Non-Disclosed              15                3.346000   

                       Total Purchase Value  Avg Total Purchase per Person  
Gender                                                                      
Female                               361.94                       4.468395  
Male                                1967.64                       4.065372  
Other / Non-Disclosed                 50.19                       4.562727

# Age Demographics

In [102]:
# Create bins in which to place values based upon TED Talk views
bins = [0, 9, 14, 19, 24, 29, 34, 39, 99999999]

# Create labels for these bins
grp_lab = ["<10", "10-14", "15-19", "20-24", "25-29", "30-34", "34-39", "40+"]

In [103]:
bins

[0, 9, 14, 19, 24, 29, 34, 39, 99999999]

In [104]:
grp_lab

['<10', '10-14', '15-19', '20-24', '25-29', '30-34', '34-39', '40+']

In [105]:
df["Age Group"] = pd.cut(df.Age, bins, labels=group_labels, include_lowest=True)
df.head()

Purchase ID             SN  Age Gender  Item ID  \
0            0        Lisim78   20   Male      108   
1            1    Lisovynya38   40   Male      143   
2            2     Ithergue48   24   Male       92   
3            3  Chamassasya86   24   Male      100   
4            4      Iskosia90   23   Male      131   

                                   Item Name  Price Age Group  
0  Extraction, Quickblade Of Trembling Hands   3.53     20-24  
1                          Frenzied Scimitar   1.56       40+  
2                               Final Critic   4.88     20-24  
3                                Blindscythe   3.27     20-24  
4                                       Fury   1.44     20-24

In [106]:
df.groupby("Age Group").Age.min()

Age Group
<10       7
10-14    10
15-19    15
20-24    20
25-29    25
30-34    30
34-39    35
40+      40
Name: Age, dtype: int64

In [107]:
cols = ["SN", "Age", "Gender", "Age Group"]

players = df.loc[:, cols]
players.head()

SN  Age Gender Age Group
0        Lisim78   20   Male     20-24
1    Lisovynya38   40   Male       40+
2     Ithergue48   24   Male     20-24
3  Chamassasya86   24   Male     20-24
4      Iskosia90   23   Male     20-24

In [108]:
players = players.drop_duplicates().reset_index(drop=True)
players.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 576 entries, 0 to 575
Data columns (total 4 columns):
 #   Column     Non-Null Count  Dtype   
---  ------     --------------  -----   
 0   SN         576 non-null    object  
 1   Age        576 non-null    int64   
 2   Gender     576 non-null    object  
 3   Age Group  576 non-null    category
dtypes: category(1), int64(1), object(2)
memory usage: 14.5+ KB


In [109]:
player_cnt = players.groupby("Age Group").size()
player_perc = players.groupby("Age Group").size() / len(players)

# https://stackoverflow.com/a/18062521
summ5 = pd.concat([player_cnt, player_perc], axis=1)
summ5.columns = ["Total Count", 'Percentage of Players']

summ5

Total Count  Percentage of Players
Age Group                                    
<10                 17               0.029514
10-14               22               0.038194
15-19              107               0.185764
20-24              258               0.447917
25-29               77               0.133681
30-34               52               0.090278
34-39               31               0.053819
40+                 12               0.020833

# Purchasing Analysis (age)

In [110]:
summ6 = df.groupby("Age Group").agg({"Purchase ID": "count", "Price": ["mean", "sum"]})
summ6.columns = ["Purchase Count", "Average Purchase Price", "Total Purchase Value"]
summ6["Avg Total Purchase per Person"] = summ6["Total Purchase Value"] / summ5["Total Count"]
summ6

Purchase Count  Average Purchase Price  Total Purchase Value  \
Age Group                                                                 
<10                    23                3.353478                 77.13   
10-14                  28                2.956429                 82.78   
15-19                 136                3.035956                412.89   
20-24                 365                3.052219               1114.06   
25-29                 101                2.900990                293.00   
30-34                  73                2.931507                214.00   
34-39                  41                3.601707                147.67   
40+                    13                2.941538                 38.24   

           Avg Total Purchase per Person  
Age Group                                 
<10                             4.537059  
10-14                           3.762727  
15-19                           3.858785  
20-24                           4.318062  
25-29                           3.805195  
30-34                           4.115385  
34-39                           4.763548  
40+                             3.186667

# Top Spenders

In [111]:
summ7 = df.groupby("SN").agg({"Purchase ID": "count", "Price": ["mean", "sum"]})
summ7.columns = ["Purchase Count", "Average Purchase Price", "Total Purchase Value"]
summ7.sort_values(by="Total Purchase Value", ascending=False, inplace=True)
summ7.head()

Purchase Count  Average Purchase Price  Total Purchase Value
SN                                                                       
Lisosia93                 5                3.792000                 18.96
Idastidru52               4                3.862500                 15.45
Chamjask73                3                4.610000                 13.83
Iral74                    4                3.405000                 13.62
Iskadarya95               3                4.366667                 13.10

# Most Popular Items

In [112]:
summ8 = df.groupby(["Item ID", "Item Name"]).agg({"Purchase ID": "count", "Price": ["mean", "sum"]})
summ8.columns = ["Purchase Count", "Average Purchase Price", "Total Purchase Value"]
summ8.sort_values(by="Purchase Count", ascending=False, inplace=True)
summ8.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
145     Fiery Glass Crusader                                       9   
132     Persuasion                                                 9   
108     Extraction, Quickblade Of Trembling Hands                  9   

                                                      Average Purchase Price  \
Item ID Item Name                                                              
92      Final Critic                                                4.614615   
178     Oathbreaker, Last Hope of the Breaking Storm                4.230000   
145     Fiery Glass Crusader                                        4.580000   
132     Persuasion                                                  3.221111   
108     Extraction, Quickblade Of Trembling Hands                   3.530000   

                                                      Total Purchase Value  
Item ID Item Name                                                           
92      Final Critic                                                 59.99  
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
145     Fiery Glass Crusader                                         41.22  
132     Persuasion                                                   28.99  
108     Extraction, Quickblade Of Trembling Hands                    31.77

# Most Profitable Items

In [113]:
summ8.sort_values(by="Total Purchase Value", ascending=False, inplace=True)
summ8.head()

Purchase Count  \
Item ID Item Name                                                      
92      Final Critic                                              13   
178     Oathbreaker, Last Hope of the Breaking Storm              12   
82      Nirvana                                                    9   
145     Fiery Glass Crusader                                       9   
103     Singed Scalpel                                             8   

                                                      Average Purchase Price  \
Item ID Item Name                                                              
92      Final Critic                                                4.614615   
178     Oathbreaker, Last Hope of the Breaking Storm                4.230000   
82      Nirvana                                                     4.900000   
145     Fiery Glass Crusader                                        4.580000   
103     Singed Scalpel                                              4.350000   

                                                      Total Purchase Value  
Item ID Item Name                                                           
92      Final Critic                                                 59.99  
178     Oathbreaker, Last Hope of the Breaking Storm                 50.76  
82      Nirvana                                                      44.10  
145     Fiery Glass Crusader                                         41.22  
103     Singed Scalpel                                               34.80

# Write-up

The three observable trends are:
    1) Males account for 84% of the total number of players. 
    2) Males account for $1,967.64 of the total revenue.
    3) Ages 20-24 account for 45% of the total number of player.
